server/src/test/java/org/elasticsearch/action/ingest/WriteableIngestDocumentTests.java

```diff
@@ -147,4 +152,42 @@ public void testToXContent() throws IOException {
         IngestDocument serializedIngestDocument = new IngestDocument(toXContentSource, toXContentIngestMetadata);
         assertThat(serializedIngestDocument, equalTo(serializedIngestDocument));
     }
+
+    static IngestDocument createRandomIngestDoc() {
+        XContentType xContentType = randomFrom(XContentType.values());
+        BytesReference sourceBytes = RandomObjects.randomSource(random(), xContentType);
+        Map<String, Object> randomSource = XContentHelper.convertToMap(sourceBytes, false, xContentType).v2();
+        return RandomDocumentPicks.randomIngestDocument(random(), randomSource);
+    }
+
+    @Override
+    protected boolean supportsUnknownFields() {
+        return true;
+    }
+
+    @Override
+    protected WriteableIngestDocument createTestInstance() {
+        return new WriteableIngestDocument(createRandomIngestDoc());
+    }
+
+    @Override
+    protected WriteableIngestDocument doParseInstance(XContentParser parser) {
+        return WriteableIngestDocument.fromXContent(parser);
+    }
+
+    @Override
+    protected Predicate<String> getRandomFieldsExcludeFilter() {
+        // We cannot have random fields in the _source field and _ingest field
```

Comment:
is it worth sharing this predicate somewhere given that it's the same in all these tests?

Reply:
I was thinking the same. What would you suggest?

Reply:
The same also goes for `testFromXContentWithFailures`. Probably should extend `AbstractXContentTestCase` with `AbstractXContentWithFailureTests` and add this method and `createTestInstanceWithFailures`. Then extend it everywhere.

Reply:
I would keep it simple, add a static method somewhere and call it where it's needed. On extending AbstractXContentTestCase and adding another base test class, I am not too sure, just because we have already a lot of test base classes and it's hard for folks to figure out which one is right for them. Maybe it's a good idea though, if you want to go that route I am not against it, I think :)

Reply:
Okay. But its not the same everywhere. At two places it is `field.startsWith` and the other two is `field.contains`. I can make everything contains and it would still work I guess. But `startsWith` is more accurate in some cases.

Reply:
ok, I thought that they were all the same, let's not complicate it then, I am fine with leaving as-is